java/vector/src/main/java/org/apache/arrow/vector/util/ByteFunctionHelpers.java

```diff
@@ -133,6 +145,17 @@ private static int memcmp(
       n -= 8;
     }
 
+    while (n > 3) {
+      int leftInt = PlatformDependent.getInt(lPos);
+      int rightInt = PlatformDependent.getInt(rPos);
+      if (leftInt != rightInt) {
+        return unsignedIntCompare(Integer.reverseBytes(leftInt), Integer.reverseBytes(rightInt));
```

Comment:
maybe i am missing something :) but why are we reversing the bytes here..should not PlatformDependent already return the interpreted value..

Reply:
Thanks for your comments. Seems you are right, I just make it similar with Long value logic(https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/util/ByteFunctionHelpers.java#L129).
I would like to remove these reverse operations and add UT for this class, what do you think?

Reply:
I updated this PR, would you mind taking a look again? thanks!